syntax="proto3";

import "common.proto";

package navigate;

message NavigateMsg {                           // 奇数是请求，偶数是响应
    oneof choice {
        common.TestReq  test_req        = 1;    // 请求
        common.TestRsp  test_rsp        = 2;    // 响应

        GetAccessReq    get_access_req  = 3;    // 获取接入点
        GetAccessRsp    get_access_rsp  = 4;  

        PutLoadReq      put_load_req    = 5;    // 上传负载
        PutLoadRsp      put_load_rsp    = 6;

    }
}

enum ErrorCode {
    SUCCESS                         = 0;

    // 系统错误码 (各个服务service_id * 10000 + 公共系统错误码)
    ERR_SYS_BEGIN                   = 102000100;
    ERR_SYS_OVERLOAD                = 102000101;    // 服务过载
    ERR_SYS_REJECT_SERVICE          = 102000102;    // 拒绝服务
    ERR_SYS_SERVER_INNER            = 102000103;    // 服务内部错误
    ERR_SYS_TIMEOUT                 = 102000104;    // 超时错误
    ERR_SYS_NO_INSERVICE_LIST       = 102000105;    // 没有可用服务ip
    ERR_SYS_TASK_STATE              = 102000106;    // 任务状态错误
    ERR_SYS_TASK_DISCARD            = 102000107;    // 任务丢弃
    ERR_SYS_END                     = 102000199;

    // 数据包错误码 (各个服务service_id * 10000 + 公共数据包错误码)
    ERR_PACKET_BEGIN                = 102000200;
    ERR_PACKET_ENCODE               = 102000201;   	// 打包失败
    ERR_PACKET_DECODE               = 102000202;   	// 解码失败
    ERR_PACKET_HEADER               = 102000203;
    ERR_PACKET_LEN                  = 102000204;
    ERR_PACKET_VERSION              = 102000205;
    ERR_PACKET_FROM_SERVICE_ID      = 102000206;
    ERR_PACKET_TO_SERVICE_ID        = 102000207;
    ERR_PACKET_APP_ID               = 102000208;
    ERR_PACKET_APP_VERSION          = 102000209;
    ERR_PACKET_CONN_SEQ_ID          = 102000210;
    ERR_PACKET_MSG_SEQ_ID           = 102000211;
    ERR_PACKET_DATA_FORMAT          = 102000212;
    ERR_PACKET_DATA_FIELD_0         = 102000213;
    ERR_PACKET_DATA_FIELD_1         = 102000214;
    ERR_PACKET_DATA_FIELD_2         = 102000215;
    ERR_PACKET_DATA_FIELD_3         = 102000216;
    ERR_PACKET_CHECK_SUM            = 102000217;
    ERR_PACKET_UNKNOWN_REQUEST      = 102000218;   	// 不明请求
    ERR_PACKET_END                  = 102000299;

    // 接口错误码 (各个服务service_id * 10000 + 公共接口错误码)
    ERR_INTERFACE_BEGIN             = 102000300;
    ERR_INTERFACE_PARAM             = 102000301;   	// 参数错误
    ERR_INTERFACE_PERM              = 102000302;   	// 权限错误
    ERR_INTERFACE_TIMEOUT           = 102000303;   	// 接口超时
    ERR_INTERFACE_END               = 102000399;

    // 业务错误码 (各个服务service_id * 10000 + 公共业务错误码)
    ERR_BUSINESS_BEGIN              = 102001000;
    ERR_BUSINESS_END                = 102009999;
}

message GetAccessReq {
    int32   service_id          = 1;
    uint64  user_id             = 2;
}

message GetAccessRsp {
    bytes   access_key          = 1;
    int32   service_id          = 2;
    uint32  proc_id             = 3;
    bytes   out_ip              = 4;
    int32   out_port            = 5;
}

message PutLoadReq {
    int32   service_id          = 1;
    uint32  proc_id             = 2;
    int32   load_val            = 3;
}

message PutLoadRsp {

}

